storage/rocksdb/ha_rocksdb.cc

```diff
@@ -12456,16 +12463,25 @@ int ha_rocksdb::update_write_sk(const TABLE *const table_arg,
       rc = check_partial_index_prefix(table_arg, kd, row_info.tx,
                                       row_info.old_data);
       if (!rc) {
-        row_info.tx->get_indexed_write_batch(m_tbl_def->get_table_type())
-            ->SingleDelete(kd.get_cf(), old_key_slice);
+        const auto s =
+            row_info.tx->get_indexed_write_batch(m_tbl_def->get_table_type())
+                ->SingleDelete(kd.get_cf(), old_key_slice);
+        if (!s.ok()) {
+          return row_info.tx->set_status_error(table->in_use, s, kd, m_tbl_def,
+                                               m_table_handler);
+        }
         bytes_written = old_key_slice.size();
       } else if (rc != HA_ERR_KEY_NOT_FOUND) {
         return rc;
       }
     } else {
       // Unconditionally issue SD if rocksdb_partial_index_blind_delete.
-      row_info.tx->get_indexed_write_batch(m_tbl_def->get_table_type())
-          ->SingleDelete(kd.get_cf(), old_key_slice);
+      const auto s =
+          row_info.tx->get_indexed_write_batch(m_tbl_def->get_table_type())
+              ->SingleDelete(kd.get_cf(), old_key_slice);
+      if (!s.ok()) {
+        return rdb_error_to_mysql(s);
```

Comment:
nit: maybe also change this from rdb_error_to_mysql to row_info.tx->set_status_error().